File: Frontend/read.php
<?php

require 'vendor/autoload.php'; // Include the Composer autoloader

use PhpAmqpLib\Connection\AMQPStreamConnection;
session_start();

// RabbitMQ server connection parameters
$rabbitMQHost = '10.147.17.79'; // Change this to your RabbitMQ server IP address
$rabbitMQPort = 5672; // Default RabbitMQ port
$rabbitMQUser = 'backend';
$rabbitMQPassword = 'password';
$virtualHost = '/';

// Queue name (frontend-to-backend-queue)
$queueName = 'front-reg-response'; //back-front

try {
    // Create a connection to RabbitMQ
    $connection = new AMQPStreamConnection($rabbitMQHost, $rabbitMQPort, $rabbitMQUser, $rabbitMQPassword, $virtualHost);

    // Create a channel
    $channel = $connection->channel();

    // Declare the queue
    list($queueName) = $channel->queue_declare($queueName, false, true, false, false);

    echo "Waiting for messages from the backend...\n";

    // Set a timeout of 30 seconds (adjust as needed)
    $timeout = 100;
    $start_time = time();
    ob_start();

    // Callback function to process received messages
    $callback = function ($message) {
        // Store the received message in the session
        $_SESSION['received_message'] = $message->body;

        echo 'Received: ', $message->body, "\n";

        $decodedMessage = json_decode($message->body, true);

        if ($decodedMessage !== null) {
        // Check if the 'status' field is 'success'
                if (isset($decodedMessage['status']) && $decodedMessage['status'] === 'success') {
                        // Redirect the user to the home screen or any desired location
                        header("Location: login2.php");
                        exit(); // Ensure script termination after the header redirection
                } else if (isset($decodedMessage['status']) && $decodedMessage['status'] === 'success') {
                        // Redirect the user to the home screen or any desired location
                        header("Location: index.php?error=user_exists");
                        exit(); // Ensure script termination after the header redirection
                }
                else {
                        header("Location: index.php?error=no_success");
                        exit();
                }
        } else {
                // Handle JSON decoding error (invalid JSON format)
                header("Location: index.php?error=json_decode_error");
                exit();
        }
    };

    // Consume messages from the queue
    $channel->basic_consume($queueName, '', false, true, false, false, $callback);

    // Wait for messages or until the timeout is reached
    while (count($channel->callbacks) && (time() - $start_time) < $timeout) {
        $channel->wait(null, false, $timeout);
    }

    $channel->close();
    $connection->close();

    // Check if the timeout occurred
    if ((time() - $start_time) >= $timeout) {
        ob_end_clean(); // Clean (erase) the output buffer
        header("Location: index.php?error=timeout");
        exit(); // Terminate the script
    }

} catch (\Exception $e) {
    echo "Error: " . $e->getMessage() . "\n";
}

?>

File: Database/Working Files/read.py
import pika

# RabbitMQ server connection parameters
rabbitmq_host = '10.248.179.10' #if we don’t get a static ip address this has to be changed to my ip everytime
rabbitmq_port = 5672  # Default RabbitMQ port
rabbitmq_user = 'database'
rabbitmq_password = 'password'
virtual_host = '/'

# Queue name (backend-to-frontend-queue)
queue_name = 'back-data'  # Replace with the actual queue name

def callback(ch, method, properties, body):
    print(f"Received: {body}")

try:
    # Create a connection to RabbitMQ
    credentials = pika.PlainCredentials(rabbitmq_user, rabbitmq_password)
    parameters = pika.ConnectionParameters(rabbitmq_host, rabbitmq_port, virtual_host, credentials)
    connection = pika.BlockingConnection(parameters)

    # Create a channel
    channel = connection.channel()

    # Declare the queue
    channel.queue_declare(queue=queue_name, durable=True)

    print("Waiting for messages from the backend...")

    # Set up a callback function to process received messages
    channel.basic_consume(queue=queue_name, on_message_callback=callback, auto_ack=True)

    # Start consuming messages
    channel.start_consuming()

except Exception as e:
    print(f"Error: {e}")
